Sources/Shared/Models/ConnectedApp.swift

```diff
@@ -0,0 +1,244 @@
+            String.Key.identifier: String.Value.identifier,
+        ]
+    }
+}
+
+/// A `ConnectedApp` represents a connection to a social media platform. Some activities, like simultaneously live
+/// stream to multiple destinations, or publishing across platforms, require requesting specific scopes. The scopes
+/// required will always be returned in the `neededScopes` array.
+/// - Note: Some properties are specific to a particular platform. These cases have been noted in the documentation
+///         where relevant.
+@objcMembers
+public class ConnectedApp: VIMModelObject {
+
+    /// The date when the user established this connected app, as a String.
+    /// - Note: This property is available to provide interoperability with Objective-C codebases.
+    ///         Using `addDate` is preferred.
+    public private(set) var addDateString: String?
+
+    /// The date when the user established this connected app.
+    public private(set) var addDate: Date?
+
+    /// Facebook only. A value of 1 maps to expired data access. A value of 0 maps to having data access.
+    /// - Note: This is a private property. Use `isDataAccessExpired` instead.
+    public private(set) var dataAccessIsExpired: NSNumber?
+
+    /// Returns whether the user's data access has expired.
+    /// - Note: Facebook only. Will always return false if the `ConnectedAppType` is not `.facebook`.
+    @nonobjc public lazy var isDataAccessExpired: Bool = {
+        guard self.type == .facebook else { return false }
+        return self.dataAccessIsExpired?.intValue != 0
+    }()
+
+    /// The list of remaining scopes on this connected app that the user needs for a particular Vimeo feature.
+    public private(set) var neededScopes: ConnectedAppScopes?
+
+    /// The list of third party pages associated with the user's account.
+    /// - Note: Facebook and LinkedIn only.
+    public private(set) var pages: [PublishOptionItem]?
+
+    /// The list of third party categories that can be selected when publishing to a social media platform.
+    /// - Note: Facebook and YouTube only.
+    public private(set) var publishCategories: [PublishOptionItem]?
+
+    /// The resource key string of the connected app.
+    public private(set) var resourceKey: String?
+
+    /// The unique identifier for the user on this connected app.
+    public private(set) var thirdPartyUserID: String?
+
+    /// The user's display name on this connected app.
+    public private(set) var thirdPartyUserDisplayName: String?
+
+    /// The type of the connected app, as a String.
+    /// - Note: This property is available to provide interoperability with Objective-C codebases.
+    ///         Using `type` is preferred.
+    public private(set) var typeString: String?
+
+    /// The type of the connected app.
+    /// - Note: A returned value of `nil` indicates an unsupported `ConnectedAppType`, or a malformed response.
+    @nonobjc public var type: ConnectedAppType? {
+        switch self.typeString {
```

Comment:
What do you think of moving this to the `ConnectedAppType` instead? Does it feel weird to conform to RawRepresentable `String` when the backing type of the enum is `Int`?

```
extension ConnectedAppType: RawRepresentable {
    public typealias RawValue = String
    public var rawValue: RawValue { return String(describing: self) }
    public init?(rawValue: Self.RawValue) {
        switch rawValue { ... }
    }
}
```

Reply:
That's a good suggestion, thanks. I've change the `type var` implementation to this:

```Swift
@nonobjc public var type: ConnectedAppType? {
    typeString.map { ConnectedAppType.init(rawValue: $0) } ?? nil
}
```

So we can still return nil if something goes wrong, or we encounter an unsupported platform. 

But the backing type for `ConnectedAppType` is an Int only for Objective-C interoperability.

Reply:
Could probably go point free `typestring.map(ConnectedAppType.init)`

Reply:
I'd still need to `?? nil` otherwise I'm trying to return an optional optional.